### PR TITLE
InitWallet panic bug

### LINKED ISSUE #612: panic on InitWallet restore:true with passphrase different than TDEX_WALLET_UNLOCK_PASSWORD_FILE set
```
time="2022-08-02T21:32:17Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:17Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:17Z" level=info msg="Replay took: 41.201µs\n"
time="2022-08-02T21:32:17Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:17Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:17Z" level=info msg="Replay took: 4.801µs\n"
time="2022-08-02T21:32:17Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:17Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:17Z" level=info msg="Replay took: 4.803µs\n"
time="2022-08-02T21:32:17Z" level=info msg="rescan internal wallet's utxo set. This could take a while..."
time="2022-08-02T21:32:17Z" level=debug msg="Restored uspents: 0"
time="2022-08-02T21:32:17Z" level=debug msg="Restore took: 0.00s"
time="2022-08-02T21:32:17Z" level=info msg=done
time="2022-08-02T21:32:17Z" level=info msg="starting daemon"
time="2022-08-02T21:32:17Z" level=info msg="wallet unlocker interface is listening on :9000"
time="2022-08-02T21:32:17Z" level=debug msg="stop grpc-web Operator server"
time="2022-08-02T21:32:17Z" level=debug msg="stop grpc Operator server"
time="2022-08-02T21:32:17Z" level=debug msg="stop mux Operator"
panic: passphrase provided is not correct

goroutine 121 [running]:
github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).startListeningToReadyChan(0xc010334000)
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:482 +0x3d0
created by github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).Start
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:277 +0x10a
time="2022-08-02T21:32:47Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:47Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:47Z" level=info msg="Replay took: 34.603µs\n"
time="2022-08-02T21:32:47Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:47Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:47Z" level=info msg="Replay took: 4.78µs\n"
time="2022-08-02T21:32:47Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:47Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:47Z" level=info msg="Replay took: 9.42µs\n"
time="2022-08-02T21:32:47Z" level=info msg="rescan internal wallet's utxo set. This could take a while..."
time="2022-08-02T21:32:47Z" level=debug msg="Restored uspents: 0"
time="2022-08-02T21:32:47Z" level=debug msg="Restore took: 0.00s"
time="2022-08-02T21:32:47Z" level=info msg=done
time="2022-08-02T21:32:47Z" level=info msg="starting daemon"
time="2022-08-02T21:32:47Z" level=info msg="wallet unlocker interface is listening on :9000"
time="2022-08-02T21:32:47Z" level=debug msg="stop grpc-web Operator server"
time="2022-08-02T21:32:47Z" level=debug msg="stop grpc Operator server"
time="2022-08-02T21:32:47Z" level=debug msg="stop mux Operator"
panic: passphrase provided is not correct

goroutine 97 [running]:
github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).startListeningToReadyChan(0xc00062c480)
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:482 +0x3d0
created by github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).Start
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:277 +0x10a
```

### PR DESCRIPTION
**Pull request by sekulicd**
This removes panic that was happening as reported in issue #612 

This closes #612 

@tiero @altafan please review.

### CODE REVIEW

File: internal/interfaces/grpc/service.go [hunk 1]
```diff
@@ -520,12 +522,29 @@ func (s *service) startListeningToReadyChan() {
 	}
 
 	if s.walletPassword != "" {
-		if err := s.opts.WalletUnlockerSvc.UnlockWallet(context.Background(), s.walletPassword); err != nil {
-			panic(err)
-		}
+		retryAttempt := 0
+	loop:
```
Comment (altafan) [inline internal/interfaces/grpc/service.go:526]: this label is actually not needed

File: internal/interfaces/grpc/service.go [hunk 2]
```diff
@@ -520,12 +522,29 @@ func (s *service) startListeningToReadyChan() {
...
+		for {
+			if err := s.opts.WalletUnlockerSvc.UnlockWallet(
+				context.Background(),
+				s.walletPassword,
+			); err != nil && err != wallet.ErrInvalidPassphrase && retryAttempt < 5 {
+				retryAttempt++
+				log.Debugf("retrying to unlock wallet: %v, attempt: %v", err, retryAttempt)
+
+				continue loop
```
Comment (altafan) [inline internal/interfaces/grpc/service.go:535]:
```suggestion
				continue
```

File: internal/interfaces/grpc/service.go [hunk 3]
```diff
@@ -520,12 +522,29 @@ func (s *service) startListeningToReadyChan() {
...
+					log.WithError(err).Warn(
+						"an error occurred while unlocking wallet",
+					)
+				} else {
+					log.Debug("wallet unlocked with password file")
+				}
 
-		s.walletPassword = ""
-		log.Infoln("wallet unlocked")
+				break loop
```
Comment (altafan) [inline internal/interfaces/grpc/service.go:545]:
```suggestion
				break
```

File: internal/interfaces/grpc/service.go (hunk 1 above)
Comment (sekulicd) [reply internal/interfaces/grpc/service.go]: Done

File: internal/interfaces/grpc/service.go (hunk 2 above)
Comment (sekulicd) [reply internal/interfaces/grpc/service.go]: Done

File: internal/interfaces/grpc/service.go (hunk 3 above)
Comment (sekulicd) [reply internal/interfaces/grpc/service.go]: Done

**Review by altafan — APPROVED** [review]

**Comment by altafan** [comment]
tACK cb14c2ec2aa196368e3fb7b3c18643eac326fe80

**Review by altafan — APPROVED** [review]
tACK cb14c2ec2aa196368e3fb7b3c18643eac326fe80

**Pull request merged**